[Bug]: Playback stops at 59s and app closes unexpectedly

### Before submitting

- [x] I searched the open and closed issues for this problem.
- [x] I can reproduce this on the latest available Flow release or a current development build.

### Flow version

2.2.0 with latest nightly

### Android version

Android 16

### Device

Oneplus 15

### Installation source

GitHub release APK

### Affected area

Video playback

### Reproduction steps

Today I put on a playlist and after a few songs it stopped playing. I opened the app and saw that it had stopped at 59 seconds and then closed straight away. I don’t have an error report to share. I’m sorry. 

### Expected behavior

The playlist should keep playing continuously without stopping or closing the app.

### Actual behavior

Playback stopped at approximately 59 seconds into a track, and the app closed immediately afterward. No error log was generated.

### Relevant media or URL

_No response_

### Logs

```shell

```

### Additional context

_No response_

============================================================
FLOW DIAGNOSTICS REPORT
Generated: 2026-08-18 11:41:20
============================================================

Manufacturer : OnePlus
Model        : CPH2747
Android      : 16 (SDK 36)
Brand        : OnePlus
App version  : 2.2.0 (17)

============================================================
SESSION LOGS  (W/E level, current session)
============================================================
--------- beginning of system
08-18 11:39:48.714  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(0, 0), format:-1, blastBufferQueue:null
08-18 11:39:48.765  1248  1248 W VRI[IconFlowRed]: handleResized, msg:, frameChanged:false, configChanged:false, displayChanged:false, attachedFrameChanged:false, compatScaleChanged:false, pendingDragResizing=false
08-18 11:39:56.298  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-1, blastBufferQueue:android.graphics.BLASTBufferQueue@6761495
08-18 11:39:56.304  1248  1248 E OplusCustomizeRestrictionManager: sInstance is null, start a new sInstance
08-18 11:39:56.363  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@6761495
08-18 11:39:56.400  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@6761495
--------- beginning of main
08-18 11:40:49.245  1248 14603 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21547s left
08-18 11:40:49.287  1248 14603 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21547s left
08-18 11:40:50.488  1248 14603 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21546s left
08-18 11:40:52.676  1248 14603 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21544s left
08-18 11:40:56.136  1248 14603 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21540s left
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal: Playback error
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:416)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:   Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:40:57.268  1248 14291 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:1572)
08-18 11:40:57.277  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=2 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: ExoPlayer - onPlayerError() called with:
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loop(Looper.java:416)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:40:57.278  1248  1248 E PlayerErrorHandler: 	at java.lang.Thread.run(Thread.java:1572)
08-18 11:40:57.279  1248  1248 W QualityManager: Stream marked as failed: https://rr2---sn-vg5obxn25po-8ajz.googlevideo.com/videoplayback?expire=1787067596&ei=bCiEar6XMc2KoccP5rbVkAQ&ip=90.167.21.31&id=o-AHt3DbYnBn0De0SIftrHGmbCxn4x3wBuWM3JuMmJ9YoW&itag=398&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&cps=421&met=1787045996%2C&mh=A-&mm=31%2C29&mn=sn-vg5obxn25po-8ajz%2Csn-ajh5q5-5a&ms=au%2Crdu&mv=m&mvi=2&pl=24&rms=au%2Cau&initcwndbps=1818750&bui=AR3QkAm1AljsWSZJ8HpNrMxvae9SDkB_iMFVD23W9aUadBcr1YZ92iKhfJm9X1dXYZq40dTQa1enALCv&spc=KBGBcvrLG1dqWRa6VOiDGVwNiAoHDyK-wlK80XewRw&vprv=1&svpuc=1&mime=video%2Fmp4&rqh=1&gir=yes&clen=50590721&dur=395.361&lmt=1767458048772586&mt=1787045585&fvip=4&keepalive=yes&fexp=51565116%2C51946838&c=ANDROID_VR&txp=5537534&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AE0s2JYwRQIgJU2MCERGE1qt-gWxpWFCzBjw3HgcdQAEsZhDSb11nbwCIQCMNhvvf7ssfPzGPZGOoW6bY8cCIKN02_zr8ZARqyR0bg%3D%3D&lsparams=cps%2Cmet%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRAIgG-eIoA3jeXUxkq8r94Ke9hjZu4zS8GTfFe2VoDdkYyYCIG8SMCx5PTW6quNA1fKkPEb4v6ZkkbUgruwWpmmzZnxc&pot=MvkEEmWvmg0q678SWgBSbZkBiUvJ_53g9lEvV9vLL-WJ5mv7Z5ZXlP5OoZQQ3pun8w0IdH92QGbcsTBCk_oV6AU_F4KASnHXK7VUma28qs1MO8uDtNIO6TKWeH_Qdi-w9Z6ZDkbwrmPO2PlB_7r8W39MZiB0GAiUBNpWt7N0wt1xz8yf2mKQzxrQ1VVMoMYlxKrVZtez0SUea3hpEx-tjEnXW9xl6FK_Jc8FkeVsYHeXobSCDR2ZsfPIu29XZ3Xed-8La0oYuEBeaVPiYeD7jQtX1Ii9r35__9wcpyahdj0X3n1eIp-u_Xd4w1az-8YWcGajPa_u3I1QiXkeUtGrtyJXauJ9hmuOGUVCroLUWK0iY8qy7pnL_Dp-DsiKDbZkFiQ3v0jowoLCVtS9615et4IlthdgDc1ykecF0zxPbpieP_3ldnvGX874PqTlcGFHV8hfjV9K4nz4lip-nJvMwnhLSXRvHKTvQk2GfM4eFbWDZbFrEYDxzoG34WUbuAcgfHtmrCkefm3LpfJenhz_W3hqZX9mSgTGcFNlzDHklaJ59JJWVBtMGTQjdKtbgKHb7dlPlzTqTBtMQVzYFnJiF87Moy5EBQytTSy8-gAcs454w0Jf4o3zrH8s24JAt8zxKQnc5SF7MwL43vMfQx3VXMTbfGULY2HReJ1Zax4ydvk77AoQu2s22ybpa8VGw7Gj6E_vDh_j9JnU3JdsP-HWruSQVpw2ih2nafJcYB-y8zOX6YCEQEHhjoHE-SsBDMx3jhQy1VrG8QrG1ACs22ncGOF-VZXJlqKDHVvoPjsePkLqCVYJccnjXhmOWKRP56Ct44yvfm3q3vvb1D30 - Error count: 1
08-18 11:40:57.279  1248  1248 W QualityManager: Codec fallback: AV1 720p gated by CDN (403) → H264 720p
08-18 11:40:57.280  1248  1248 W FlowVideoAutoNext: loadMediaInternal audioOnly=false preserve=59860 local=false | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.280  1248  1248 W FlowVideoAutoNext: clearPreload | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.280  1248  1248 W MediaLoader: Surface not ready yet, preparing media and waiting for attach
08-18 11:40:57.284  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.284  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.284  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.284  1248  1248 W PlayerErrorHandler: HTTP 403 on AV1 — switched to a compatible codec at the same resolution
08-18 11:40:57.285  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.285  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.285  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.285  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.285  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.285  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.285  1248  1248 W FlowVideoAutoNext: onIsPlayingChanged isPlaying=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.286  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.288  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.288  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=15cEtTMmvJk | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.289  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.289  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.289  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.289  1248  1248 W FlowVideoAutoNext: onPlaybackStateChanged BUFFERING recovering=false live=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=59860/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.297  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:40:57.299  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.299  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.299  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.299  1248 14291 W libc    : Access denied finding property "vendor.debug.stagefright.mediacodec.trace"
08-18 11:40:57.299  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:40:57.299  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:40:57.299  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:40:57.299  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:40:57.300  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.300  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.300  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:40:57.303  1248 15165 W thub.aedev.flow: AIBinder_linkToDeath is being called with a non-null cookie and no onUnlink callback set. This might not be intended. AIBinder_DeathRecipient_setOnUnlinked should be called first.
08-18 11:40:57.307  1248 15165 E thub.aedev.flow: Failed to query component interface for required system resources: 6
08-18 11:40:57.309  1248 15165 W thub.aedev.flow: AIBinder_linkToDeath is being called with a non-null cookie and no onUnlink callback set. This might not be intended. AIBinder_DeathRecipient_setOnUnlinked should be called first.
08-18 11:40:57.495  1248 15164 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21539s left
08-18 11:40:57.696  1248 15164 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21539s left
08-18 11:40:58.895  1248 15164 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21538s left
08-18 11:41:01.143  1248 15164 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=970995+161781 expire=valid 21535s left
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal: Playback error
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:416)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:   Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:41:01.149  1248 14291 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:1572)
08-18 11:41:01.156  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=1 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: ExoPlayer - onPlayerError() called with:
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loop(Looper.java:416)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:41:01.157  1248  1248 E PlayerErrorHandler: 	at java.lang.Thread.run(Thread.java:1572)
08-18 11:41:01.158  1248  1248 W QualityManager: Stream marked as failed: https://rr2---sn-vg5obxn25po-8ajz.googlevideo.com/videoplayback?expire=1787067596&ei=bCiEar6XMc2KoccP5rbVkAQ&ip=90.167.21.31&id=o-AHt3DbYnBn0De0SIftrHGmbCxn4x3wBuWM3JuMmJ9YoW&itag=298&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&cps=421&met=1787045996%2C&mh=A-&mm=31%2C29&mn=sn-vg5obxn25po-8ajz%2Csn-ajh5q5-5a&ms=au%2Crdu&mv=m&mvi=2&pl=24&rms=au%2Cau&initcwndbps=1818750&bui=AR3QkAm1AljsWSZJ8HpNrMxvae9SDkB_iMFVD23W9aUadBcr1YZ92iKhfJm9X1dXYZq40dTQa1enALCv&spc=KBGBcvrLG1dqWRa6VOiDGVwNiAoHDyK-wlK80XewRw&vprv=1&svpuc=1&mime=video%2Fmp4&rqh=1&gir=yes&clen=36021909&dur=395.361&lmt=1767458023288940&mt=1787045585&fvip=4&keepalive=yes&fexp=51565116%2C51946838&c=ANDROID_VR&txp=5535534&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AE0s2JYwRAIgQSW0hAjUU89qEX1A0Wyaypbg76PUdDZWrRJrI3hdUXwCIBmMJFKq_c_mYX2Ptnn4-DRYUxgS8kHLhO78LVgmyGxH&lsparams=cps%2Cmet%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRAIgG-eIoA3jeXUxkq8r94Ke9hjZu4zS8GTfFe2VoDdkYyYCIG8SMCx5PTW6quNA1fKkPEb4v6ZkkbUgruwWpmmzZnxc&pot=MvkEEmWvmg0q678SWgBSbZkBiUvJ_53g9lEvV9vLL-WJ5mv7Z5ZXlP5OoZQQ3pun8w0IdH92QGbcsTBCk_oV6AU_F4KASnHXK7VUma28qs1MO8uDtNIO6TKWeH_Qdi-w9Z6ZDkbwrmPO2PlB_7r8W39MZiB0GAiUBNpWt7N0wt1xz8yf2mKQzxrQ1VVMoMYlxKrVZtez0SUea3hpEx-tjEnXW9xl6FK_Jc8FkeVsYHeXobSCDR2ZsfPIu29XZ3Xed-8La0oYuEBeaVPiYeD7jQtX1Ii9r35__9wcpyahdj0X3n1eIp-u_Xd4w1az-8YWcGajPa_u3I1QiXkeUtGrtyJXauJ9hmuOGUVCroLUWK0iY8qy7pnL_Dp-DsiKDbZkFiQ3v0jowoLCVtS9615et4IlthdgDc1ykecF0zxPbpieP_3ldnvGX874PqTlcGFHV8hfjV9K4nz4lip-nJvMwnhLSXRvHKTvQk2GfM4eFbWDZbFrEYDxzoG34WUbuAcgfHtmrCkefm3LpfJenhz_W3hqZX9mSgTGcFNlzDHklaJ59JJWVBtMGTQjdKtbgKHb7dlPlzTqTBtMQVzYFnJiF87Moy5EBQytTSy8-gAcs454w0Jf4o3zrH8s24JAt8zxKQnc5SF7MwL43vMfQx3VXMTbfGULY2HReJ1Zax4ydvk77AoQu2s22ybpa8VGw7Gj6E_vDh_j9JnU3JdsP-HWruSQVpw2ih2nafJcYB-y8zOX6YCEQEHhjoHE-SsBDMx3jhQy1VrG8QrG1ACs22ncGOF-VZXJlqKDHVvoPjsePkLqCVYJccnjXhmOWKRP56Ct44yvfm3q3vvb1D30 - Error count: 1
08-18 11:41:01.158  1248  1248 W PlayerErrorHandler: HTTP 403 — stream URL expired ('YouTube changed data'). Triggering extractor reload.
08-18 11:41:01.158  1248  1248 W PlayerErrorHandler: Stream expired - requesting full extractor reload (attempt 1/3). reason=http-403 video=720p codec=h264 videoItag=298 videoMime=video/mp4 audioItag=140 audioMime=audio/mp4
08-18 11:41:01.158  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.158  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.159  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.159  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.166  1248  1248 W VideoPlayerViewModel: Stream expired — re-fetching streams for 15cEtTMmvJk (attempt 1/3)
08-18 11:41:01.166  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.166  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.168  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=false playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.168  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=false playing=false pos=59860/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:01.169  1248  1248 W FlowVideoAutoNext: onPlayWhenReadyChanged playWhenReady=false reason=1 | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=59860/395361 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:01.170  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=0 | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:01.171  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=null | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:01.174  1248 14099 W InnerTubeVideoExtractor: Extraction start for 15cEtTMmvJk (forceSabr=true)
08-18 11:41:02.135  1248 14088 W NewPipeExtractor: NewPipe nsig ineffective (unchanged) — disabling for this session, falling back to home-grown
08-18 11:41:02.491  1248 14088 W Flow_CipherFnExtract: No sig pattern matched, checking for Q-array obfuscation...
08-18 11:41:02.502  1248 14088 E Flow_CipherFnExtract: ========== SIG FUNCTION EXTRACTION FAILED ==========
08-18 11:41:02.502  1248 14088 E Flow_CipherFnExtract: Could not find signature deobfuscation function name
08-18 11:41:02.661  1248 14088 W Flow_CipherDeobfusc: Could not extract signature function info from player JS; n-transform may still work
08-18 11:41:02.857  1248 14088 W InnerTubeVideoExtractor: WEB+PoToken (SABR) resolved: 20 video (144, 240, 360, 480, 720, 1080p), 4 audio, sabr=true
08-18 11:41:02.857  1248 14088 W InnerTubeVideoExtractor: Extraction OK for 15cEtTMmvJk via WEB (mode=SABR/forced)
08-18 11:41:02.860  1248 14153 W VideoPlayerViewModel: VOD fallback for 15cEtTMmvJk via InnerTube (NewPipe StreamInfo null)
08-18 11:41:02.860  1248  1248 W BackgroundServiceMgr: Video MediaSessionService controller already connected for 15cEtTMmvJk
08-18 11:41:02.863  1248  1248 W FlowVideoAutoNext: setAutoplayCandidates source=15cEtTMmvJk input=0 filtered=20 enabled=false | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:02.863  1248  1248 W FlowVideoAutoNext: requestPreloadNext reason=autoplay-candidates target=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:02.863  1248  1248 W FlowVideoAutoNext: schedulePreloadNext skipped no current media item | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:41:02.863  1248  1248 W VideoPlayerViewModel: VOD fallback playing 15cEtTMmvJk via InnerTube WEB (sabr=true, video=0, audio=0)
08-18 11:41:02.864  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:02.864  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:02.864  1248  1248 W FlowVideoAutoNext: setVideoTracksDisabled disabled=false | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.864  1248  1248 W EnhancedPlayerManager: setStreams: no separate audio stream for 15cEtTMmvJk; attempting video-only/muxed playback
08-18 11:41:02.866  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:02.866  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:02.866  1248  1248 W FlowVideoAutoNext: loadMediaInternal audioOnly=false preserve=59860 local=false | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.866  1248  1248 W FlowVideoAutoNext: clearPreload | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.866  1248  1248 W MediaLoader: Surface not ready yet, preparing media and waiting for attach
08-18 11:41:02.867  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.869  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=15cEtTMmvJk | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.870  1248  1248 W FlowVideoAutoNext: onPlaybackStateChanged BUFFERING recovering=false live=false | video=15cEtTMmvJk exo=BUFFERING pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.870  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:02.870  1248  1248 W VideoPlayerService: updateLocks active=false | exo=BUFFERING pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:02.871  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.871  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.871  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.871  1248  1248 W FlowVideoAutoNext: onPlayWhenReadyChanged playWhenReady=true reason=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.872  1248  1248 W FlowVideoAutoNext: setAutoplayCandidates source=15cEtTMmvJk input=20 filtered=20 enabled=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.872  1248  1248 W FlowVideoAutoNext: requestPreloadNext reason=autoplay-candidates target=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.872  1248  1248 W FlowVideoAutoNext: schedulePreloadNext start next=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.872  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.872  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.873  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.879  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:02.880  1248 14091 W InnerTubeVideoExtractor: Extraction start for JgeKzO5xSUE (forceSabr=false)
08-18 11:41:02.880  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.880  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.881  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.884  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.884  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:02.884  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:03.027  1248 14095 W InnerTubeVideoExtractor: ANDROID_VR: BOT_WALL, reason=Sign in to confirm you’re not a bot
08-18 11:41:03.204  1248 14106 W InnerTubeVideoExtractor: ANDROID_VR: BOT_WALL, reason=Sign in to confirm you’re not a bot
08-18 11:41:03.432  1248 14091 W InnerTubeVideoExtractor: Extraction OK for JgeKzO5xSUE via ANDROID_VR (mode=DIRECT)
08-18 11:41:04.635  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=2 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:04.637  1248  1248 W FlowVideoAutoNext: schedulePreloadNext appended next=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:41:04.644  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:04.644  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:04.644  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:41:06.293  1248  1248 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
08-18 11:41:06.293  1248  1248 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
08-18 11:41:06.294  1248  1248 W FlowVideoLifecycle: onResume | interactive=true lifecycle=STARTED pip=false pendingAutoPip=false bgPref=true shortsBgPref=false explicitBg=false video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false buffering=true pos=0/-9223372036854775807 idx=0 count=2
08-18 11:41:06.305  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:null
08-18 11:41:06.307  1248  1248 W FlowVideoLifecycle: surfaceAttached video=15cEtTMmvJk pos=0 pwr=true force=true
08-18 11:41:06.332  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@4d60982
08-18 11:41:06.788  1248  1248 W Choreographer: Frame time is 0.185451 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
08-18 11:41:07.414  1248 14106 W SabrStreamCtrl: Stream protection: attestation pending (PoToken being verified)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal: Playback error
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:508)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:416)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:   Caused by: androidx.media3.exoplayer.source.UnrecognizedInputFormatException: None of the available extractors (FlvExtractor, FlacExtractor, WavExtractor, FragmentedMp4Extractor, Mp4Extractor, AmrExtractor, PsExtractor, OggExtractor, TsExtractor, MatroskaExtractor, AdtsExtractor, Ac3Extractor, Ac4Extractor, Mp3Extractor, AviExtractor, JpegExtractor, PngExtractor, WebpExtractor, BmpExtractor, HeifExtractor, AvifExtractor) could read the stream. {contentIsMalformed=false, dataType=1}
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:   sniff failures: [NoDeclaredBrand, NoDeclaredBrand]
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.BundledExtractorsAdapter.init(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:187)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.ProgressiveMediaPeriod$ExtractingLoadable.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:190)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:41:07.542  1248 14291 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:1572)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: ExoPlayer - onPlayerError() called with:
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:508)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loop(Looper.java:416)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: Caused by: androidx.media3.exoplayer.source.UnrecognizedInputFormatException: None of the available extractors (FlvExtractor, FlacExtractor, WavExtractor, FragmentedMp4Extractor, Mp4Extractor, AmrExtractor, PsExtractor, OggExtractor, TsExtractor, MatroskaExtractor, AdtsExtractor, Ac3Extractor, Ac4Extractor, Mp3Extractor, AviExtractor, JpegExtractor, PngExtractor, WebpExtractor, BmpExtractor, HeifExtractor, AvifExtractor) could read the stream. {contentIsMalformed=false, dataType=1}
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: sniff failures: [NoDeclaredBrand, NoDeclaredBrand]
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.BundledExtractorsAdapter.init(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:187)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.ProgressiveMediaPeriod$ExtractingLoadable.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:190)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: 	at java.lang.Thread.run(Thread.java:1572)
08-18 11:41:07.543  1248  1248 E PlayerErrorHandler: Source validation error: 3003 - Source error
08-18 11:41:07.543  1248  1248 W PlayerErrorHandler: Unrecognized format error — trying alternative stream format
08-18 11:41:07.543  1248  1248 E QualityManager: No working streams available - all streams failed
08-18 11:41:07.543  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:07.543  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:07.544  1248  1248 W FlowVideoAutoNext: clearPreload | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=true live=false
08-18 11:41:07.544  1248  1248 W PlayerErrorHandler: Playback shutdown initiated
08-18 11:41:07.544  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:07.544  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:41:07.544  1248  1248 E PlayerErrorHandler: Player error (notify): Source error
08-18 11:41:07.545  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=true live=false
08-18 11:41:07.547  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=0 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=true live=false
08-18 11:41:07.547  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=null | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=true live=false
08-18 11:41:14.062  1248  1248 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
08-18 11:41:14.062  1248  1248 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.

Twice in a row. This time after 1 minute and 9 seconds. 

============================================================
FLOW DIAGNOSTICS REPORT
Generated: 2026-08-18 11:43:09
============================================================

Manufacturer : OnePlus
Model        : CPH2747
Android      : 16 (SDK 36)
Brand        : OnePlus
App version  : 2.2.0 (17)

============================================================
SESSION LOGS  (W/E level, current session)
============================================================
--------- beginning of system
08-18 11:39:48.714  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(0, 0), format:-1, blastBufferQueue:null
08-18 11:39:48.765  1248  1248 W VRI[IconFlowRed]: handleResized, msg:, frameChanged:false, configChanged:false, displayChanged:false, attachedFrameChanged:false, compatScaleChanged:false, pendingDragResizing=false
08-18 11:39:56.298  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-1, blastBufferQueue:android.graphics.BLASTBufferQueue@6761495
08-18 11:39:56.304  1248  1248 E OplusCustomizeRestrictionManager: sInstance is null, start a new sInstance
08-18 11:39:56.363  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@6761495
08-18 11:39:56.400  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@6761495
08-18 11:41:06.305  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:null
08-18 11:41:06.332  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@4d60982
--------- beginning of main
08-18 11:42:46.967  1248 15879 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=1132776+161444 expire=valid 21530s left
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal: Playback error
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:416)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:   Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:42:48.290  1248 14291 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:1572)
08-18 11:42:48.298  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=2 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: ExoPlayer - onPlayerError() called with:
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loop(Looper.java:416)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:42:48.300  1248  1248 E PlayerErrorHandler: 	at java.lang.Thread.run(Thread.java:1572)
08-18 11:42:48.301  1248  1248 W QualityManager: Stream marked as failed: https://rr2---sn-vg5obxn25po-8ajz.googlevideo.com/videoplayback?expire=1787067696&ei=0CiEav61K6r5xN8P3sD3wAg&ip=90.167.21.31&id=o-AFwRG6i8cEAONwS26icMJs2hc29sIwRVE00M4E7vswQH&itag=398&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&cps=390&met=1787046096%2C&mh=A-&mm=31%2C29&mn=sn-vg5obxn25po-8ajz%2Csn-ajh5q5-5a&ms=au%2Crdu&mv=m&mvi=2&pl=24&rms=au%2Cau&initcwndbps=1818750&bui=AR3QkAnAW9DfkuIk4NCzbx49AUiJqlAsLAqPbO4k5SLeW-xAmu0omBsYWsVjl8TY7wRrQEeT4ZY-NKbi&spc=KBGBcvGzyHLRV5BLbeK_qsQBnKS7i9YYQdl8xVDL_g&vprv=1&svpuc=1&mime=video%2Fmp4&rqh=1&gir=yes&clen=50590721&dur=395.361&lmt=1767458048772586&mt=1787045585&fvip=4&keepalive=yes&fexp=51565116%2C51946838&c=ANDROID_VR&txp=5537534&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AE0s2JYwRQIhAI2ouyMshVqya5GMiWir1Sdd_QZCSHfHVHrYu1Dx_iy5AiAQlTCNFiEx78gQqtY9RHRwUun8RsR4Ftjmy5NzhTm1bQ%3D%3D&lsparams=cps%2Cmet%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRAIgc1vtZfO5MBv_MohFi5Zn8B7AhwMqcutlGI-jIOf8WkICIB62PI8PPb2ANyWib4yffbmipb7yO5B5l-hCgARxGLO8&pot=MvkEEmWvmg0q678SWgBSbZkBiUvJ_53g9lEvV9vLL-WJ5mv7Z5ZXlP5OoZQQ3pun8w0IdH92QGbcsTBCk_oV6AU_F4KASnHXK7VUma28qs1MO8uDtNIO6TKWeH_Qdi-w9Z6ZDkbwrmPO2PlB_7r8W39MZiB0GAiUBNpWt7N0wt1xz8yf2mKQzxrQ1VVMoMYlxKrVZtez0SUea3hpEx-tjEnXW9xl6FK_Jc8FkeVsYHeXobSCDR2ZsfPIu29XZ3Xed-8La0oYuEBeaVPiYeD7jQtX1Ii9r35__9wcpyahdj0X3n1eIp-u_Xd4w1az-8YWcGajPa_u3I1QiXkeUtGrtyJXauJ9hmuOGUVCroLUWK0iY8qy7pnL_Dp-DsiKDbZkFiQ3v0jowoLCVtS9615et4IlthdgDc1ykecF0zxPbpieP_3ldnvGX874PqTlcGFHV8hfjV9K4nz4lip-nJvMwnhLSXRvHKTvQk2GfM4eFbWDZbFrEYDxzoG34WUbuAcgfHtmrCkefm3LpfJenhz_W3hqZX9mSgTGcFNlzDHklaJ59JJWVBtMGTQjdKtbgKHb7dlPlzTqTBtMQVzYFnJiF87Moy5EBQytTSy8-gAcs454w0Jf4o3zrH8s24JAt8zxKQnc5SF7MwL43vMfQx3VXMTbfGULY2HReJ1Zax4ydvk77AoQu2s22ybpa8VGw7Gj6E_vDh_j9JnU3JdsP-HWruSQVpw2ih2nafJcYB-y8zOX6YCEQEHhjoHE-SsBDMx3jhQy1VrG8QrG1ACs22ncGOF-VZXJlqKDHVvoPjsePkLqCVYJccnjXhmOWKRP56Ct44yvfm3q3vvb1D30 - Error count: 1
08-18 11:42:48.302  1248  1248 W QualityManager: Codec fallback: AV1 720p gated by CDN (403) → H264 720p
08-18 11:42:48.302  1248  1248 W FlowVideoAutoNext: loadMediaInternal audioOnly=false preserve=69845 local=false | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.302  1248  1248 W FlowVideoAutoNext: clearPreload | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.302  1248  1248 W MediaLoader: Surface not ready yet, preparing media and waiting for attach
08-18 11:42:48.305  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.306  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.306  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.306  1248  1248 W PlayerErrorHandler: HTTP 403 on AV1 — switched to a compatible codec at the same resolution
08-18 11:42:48.306  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.307  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.307  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.307  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.307  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.307  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.307  1248  1248 W FlowVideoAutoNext: onIsPlayingChanged isPlaying=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.308  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.310  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.310  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=15cEtTMmvJk | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.313  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.313  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.313  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.313  1248  1248 W FlowVideoAutoNext: onPlaybackStateChanged BUFFERING recovering=false live=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=69845/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.318  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:48.320  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.320  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.320  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.322  1248 14291 W libc    : Access denied finding property "vendor.debug.stagefright.mediacodec.trace"
08-18 11:42:48.323  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:42:48.323  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:42:48.323  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:42:48.323  1248 14291 W libc    : Access denied finding property "persist.vendor.media.c2.frameworks.buffercount.trace"
08-18 11:42:48.325  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.325  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.325  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:48.327  1248 16486 W thub.aedev.flow: AIBinder_linkToDeath is being called with a non-null cookie and no onUnlink callback set. This might not be intended. AIBinder_DeathRecipient_setOnUnlinked should be called first.
08-18 11:42:48.334  1248 16486 E thub.aedev.flow: Failed to query component interface for required system resources: 6
08-18 11:42:48.336  1248 16486 W thub.aedev.flow: AIBinder_linkToDeath is being called with a non-null cookie and no onUnlink callback set. This might not be intended. AIBinder_DeathRecipient_setOnUnlinked should be called first.
08-18 11:42:48.471  1248 16485 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=1132776+161444 expire=valid 21528s left
08-18 11:42:48.606  1248 16485 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=1132776+161444 expire=valid 21528s left
08-18 11:42:49.821  1248 16485 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=1132776+161444 expire=valid 21527s left
08-18 11:42:51.974  1248 16485 W YouTubeHttpDataSource: HTTP 403 c=ANDROID_VR itag=140 mime=audio/mp4 pot=true range=1132776+161444 expire=valid 21525s left
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal: Playback error
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:416)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:   Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:42:51.983  1248 14291 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:1572)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: Handler (android.media.MediaCodec$EventHandler) {3d8586d} sending message to a Handler on a dead thread
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: java.lang.IllegalStateException: Handler (android.media.MediaCodec$EventHandler) {3d8586d} sending message to a Handler on a dead thread
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.os.MessageQueue.enqueueMessageLegacy(MessageQueue.java:1414)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.os.MessageQueue.enqueueMessage(MessageQueue.java:1563)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.enqueueMessage(Handler.java:796)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.sendMessageAtTime(Handler.java:745)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.sendMessageDelayed(Handler.java:715)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.sendMessage(Handler.java:653)
08-18 11:42:51.985  1248 16486 W LegacyMessageQueue: 	at android.media.MediaCodec.postEventFromNative(MediaCodec.java:6046)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: Handler (android.media.MediaCodec$EventHandler) {3d8586d} sending message to a Handler on a dead thread
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: java.lang.IllegalStateException: Handler (android.media.MediaCodec$EventHandler) {3d8586d} sending message to a Handler on a dead thread
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.os.MessageQueue.enqueueMessageLegacy(MessageQueue.java:1414)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.os.MessageQueue.enqueueMessage(MessageQueue.java:1563)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.enqueueMessage(Handler.java:796)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.sendMessageAtTime(Handler.java:745)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.sendMessageDelayed(Handler.java:715)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.os.Handler.sendMessage(Handler.java:653)
08-18 11:42:51.986  1248 16486 W LegacyMessageQueue: 	at android.media.MediaCodec.postEventFromNative(MediaCodec.java:6046)
08-18 11:42:51.989  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=1 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: ExoPlayer - onPlayerError() called with:
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:478)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loop(Looper.java:416)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: Caused by: androidx.media3.datasource.HttpDataSource$InvalidResponseCodeException: Response code: 403
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.okhttp.OkHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:136)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at io.github.aedev.flow.player.datasource.YouTubeHttpDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:93)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.DefaultDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:160)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.TeeDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:3)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.openNextSource(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:275)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.cache.CacheDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:128)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.datasource.StatsDataSource.open(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:11)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.chunk.ContainerMediaChunk.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:71)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:42:51.991  1248  1248 E PlayerErrorHandler: 	at java.lang.Thread.run(Thread.java:1572)
08-18 11:42:51.991  1248  1248 W QualityManager: Stream marked as failed: https://rr2---sn-vg5obxn25po-8ajz.googlevideo.com/videoplayback?expire=1787067696&ei=0CiEar3hLvGDvdIP18_Z0QQ&ip=90.167.21.31&id=o-AED1U1ZB2ZH09R_3G-NGifjsLzzWW3daOfUabilg7H0X&itag=298&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&cps=404&met=1787046096%2C&mh=A-&mm=31%2C29&mn=sn-vg5obxn25po-8ajz%2Csn-ajh5qc-5w&ms=au%2Crdu&mv=m&mvi=2&pcm2cms=yes&pl=24&rms=au%2Cau&initcwndbps=1818750&bui=AR3QkAnSyO8NvIuC_rjnYVr-2DBR2LdaSNCU_p4wMEk18Jksc6bd-7RfXNyDIOB9DaeLWcaJzhnY8sSc&spc=KBGBcjdKjNTKOc3TAPF84OfSIP-zott43kETQeO2FbA2UCasnRInpOiXzg&vprv=1&svpuc=1&mime=video%2Fmp4&rqh=1&gir=yes&clen=36021909&dur=395.361&lmt=1767458023288940&mt=1787045585&fvip=5&keepalive=yes&fexp=51565116%2C51946838&c=VISIONOS&txp=5535534&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AE0s2JYwRQIhAKaX7N6eiQVtla9qzuZm95IDmGDe7Ucr9EzwsWKDiGWCAiB8j9cQ3FA5gsWyxERw1WPQTCAa2N3kpvqURG2bAy8Haw%3D%3D&lsparams=cps%2Cmet%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRgIhAJIYiyJ5Je0xY_5PZG_ppsNIO7_DMRDrSZ_fn5b4Xi4PAiEAtcZwp74zGTMtDx_GawDZx6_whBSdaR982E5_mSW3ESA%3D&cpn=idKJwf-_hCg4DMua - Error count: 1
08-18 11:42:51.991  1248  1248 W PlayerErrorHandler: HTTP 403 — stream URL expired ('YouTube changed data'). Triggering extractor reload.
08-18 11:42:51.991  1248  1248 W PlayerErrorHandler: Stream expired - requesting full extractor reload (attempt 1/3). reason=http-403 video=720p codec=h264 videoItag=298 videoMime=video/mp4 audioItag=140 audioMime=audio/mp4
08-18 11:42:51.992  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:51.992  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:51.993  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:51.993  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.000  1248  1248 W VideoPlayerViewModel: Stream expired — re-fetching streams for 15cEtTMmvJk (attempt 2/3)
08-18 11:42:52.001  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.001  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.003  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=false playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.003  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=false playing=false pos=69845/395361 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.003  1248  1248 W FlowVideoAutoNext: onPlayWhenReadyChanged playWhenReady=false reason=1 | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=69845/395361 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:52.005  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=0 | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:52.007  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=null | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:52.023  1248 14099 W InnerTubeVideoExtractor: Extraction start for 15cEtTMmvJk (forceSabr=true)
08-18 11:42:52.381  1248 16560 W InnerTubeVideoExtractor: WEB+PoToken (SABR) resolved: 20 video (144, 240, 360, 480, 720, 1080p), 4 audio, sabr=true
08-18 11:42:52.381  1248 16560 W InnerTubeVideoExtractor: Extraction OK for 15cEtTMmvJk via WEB (mode=SABR/forced)
08-18 11:42:52.382  1248 14144 W VideoPlayerViewModel: VOD fallback for 15cEtTMmvJk via InnerTube (NewPipe StreamInfo null)
08-18 11:42:52.383  1248  1248 W BackgroundServiceMgr: Video MediaSessionService controller already connected for 15cEtTMmvJk
08-18 11:42:52.388  1248  1248 W FlowVideoAutoNext: setAutoplayCandidates source=15cEtTMmvJk input=0 filtered=20 enabled=false | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:52.388  1248  1248 W FlowVideoAutoNext: requestPreloadNext reason=autoplay-candidates target=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:52.388  1248  1248 W FlowVideoAutoNext: schedulePreloadNext skipped no current media item | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:52.388  1248  1248 W VideoPlayerViewModel: VOD fallback playing 15cEtTMmvJk via InnerTube WEB (sabr=true, video=0, audio=0)
08-18 11:42:52.390  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.390  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.390  1248  1248 W FlowVideoAutoNext: setVideoTracksDisabled disabled=false | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.391  1248  1248 W EnhancedPlayerManager: setStreams: no separate audio stream for 15cEtTMmvJk; attempting video-only/muxed playback
08-18 11:42:52.392  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.392  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.393  1248  1248 W FlowVideoAutoNext: loadMediaInternal audioOnly=false preserve=69845 local=false | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.393  1248  1248 W FlowVideoAutoNext: clearPreload | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.393  1248  1248 W MediaLoader: Surface not ready yet, preparing media and waiting for attach
08-18 11:42:52.395  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.397  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=15cEtTMmvJk | video=15cEtTMmvJk exo=IDLE pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.399  1248  1248 W FlowVideoAutoNext: onPlaybackStateChanged BUFFERING recovering=false live=false | video=15cEtTMmvJk exo=BUFFERING pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.400  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.400  1248  1248 W VideoPlayerService: updateLocks active=false | exo=BUFFERING pwr=false playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:52.401  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.401  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.401  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.401  1248  1248 W FlowVideoAutoNext: onPlayWhenReadyChanged playWhenReady=true reason=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.405  1248  1248 W FlowVideoAutoNext: setAutoplayCandidates source=15cEtTMmvJk input=20 filtered=20 enabled=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.405  1248  1248 W FlowVideoAutoNext: requestPreloadNext reason=autoplay-candidates target=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.405  1248  1248 W FlowVideoAutoNext: schedulePreloadNext start next=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.405  1248  1248 W VideoPlayerService: playerState collect update | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.405  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.405  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.414  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=1 windows=1 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.415  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.415  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.415  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.415  1248 14109 W InnerTubeVideoExtractor: Extraction start for JgeKzO5xSUE (forceSabr=false)
08-18 11:42:52.420  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.420  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.420  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.421  1248  1248 W FlowVideoAutoNext: restoreVideoOutput deferred interactive=true surfaceValid=false | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:52.502  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.502  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.502  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=1 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:52.544  1248 14091 W InnerTubeVideoExtractor: ANDROID_VR: BOT_WALL, reason=Sign in to confirm you’re not a bot
08-18 11:42:52.704  1248 16559 W InnerTubeVideoExtractor: ANDROID_VR: BOT_WALL, reason=Sign in to confirm you’re not a bot
08-18 11:42:52.876  1248 14091 W InnerTubeVideoExtractor: Extraction OK for JgeKzO5xSUE via ANDROID_VR (mode=DIRECT)
08-18 11:42:54.049  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=2 | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:54.050  1248  1248 W FlowVideoAutoNext: schedulePreloadNext appended next=JgeKzO5xSUE fromQueue=true | video=15cEtTMmvJk exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:54.053  1248  1248 W VideoPlayerService: onStartCommand action=null | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:54.053  1248  1248 W VideoPlayerService: updateLocks active=true | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:54.053  1248  1248 W VideoPlayerService: acquireLocks | exo=BUFFERING pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=true
08-18 11:42:56.673  1248 16564 W SabrStreamCtrl: Stream protection: attestation pending (PoToken being verified)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal: Playback error
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:508)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:416)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:   Caused by: androidx.media3.exoplayer.source.UnrecognizedInputFormatException: None of the available extractors (FlvExtractor, FlacExtractor, WavExtractor, FragmentedMp4Extractor, Mp4Extractor, AmrExtractor, PsExtractor, OggExtractor, TsExtractor, MatroskaExtractor, AdtsExtractor, Ac3Extractor, Ac4Extractor, Mp3Extractor, AviExtractor, JpegExtractor, PngExtractor, WebpExtractor, BmpExtractor, HeifExtractor, AvifExtractor) could read the stream. {contentIsMalformed=false, dataType=1}
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:   sniff failures: [NoDeclaredBrand, NoDeclaredBrand]
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.BundledExtractorsAdapter.init(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:187)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.source.ProgressiveMediaPeriod$ExtractingLoadable.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:190)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:42:56.970  1248 14291 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:1572)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: ExoPlayer - onPlayerError() called with:
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: androidx.media3.exoplayer.ExoPlaybackException: Source error
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:17)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:508)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at android.os.Handler.dispatchMessage(Handler.java:112)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loopOnce(Looper.java:306)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at android.os.Looper.loop(Looper.java:416)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at android.os.HandlerThread.run(HandlerThread.java:85)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: Caused by: androidx.media3.exoplayer.source.UnrecognizedInputFormatException: None of the available extractors (FlvExtractor, FlacExtractor, WavExtractor, FragmentedMp4Extractor, Mp4Extractor, AmrExtractor, PsExtractor, OggExtractor, TsExtractor, MatroskaExtractor, AdtsExtractor, Ac3Extractor, Ac4Extractor, Mp3Extractor, AviExtractor, JpegExtractor, PngExtractor, WebpExtractor, BmpExtractor, HeifExtractor, AvifExtractor) could read the stream. {contentIsMalformed=false, dataType=1}
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: sniff failures: [NoDeclaredBrand, NoDeclaredBrand]
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.BundledExtractorsAdapter.init(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:187)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.source.ProgressiveMediaPeriod$ExtractingLoadable.load(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:190)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at androidx.media3.exoplayer.upstream.Loader$LoadTask.run(r8-map-id-88cc9a92252d921c8b33ec7197f5e14aa1291c70f3f08f3bfd3d5f30673ca351:35)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1100)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: 	at java.lang.Thread.run(Thread.java:1572)
08-18 11:42:56.972  1248  1248 E PlayerErrorHandler: Source validation error: 3003 - Source error
08-18 11:42:56.972  1248  1248 W PlayerErrorHandler: Unrecognized format error — trying alternative stream format
08-18 11:42:56.972  1248  1248 E QualityManager: No working streams available - all streams failed
08-18 11:42:56.972  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:56.972  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:56.972  1248  1248 W FlowVideoAutoNext: clearPreload | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=2 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=JgeKzO5xSUE attempt=15cEtTMmvJk->JgeKzO5xSUE retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:56.972  1248  1248 W PlayerErrorHandler: Playback shutdown initiated
08-18 11:42:56.973  1248  1248 W VideoPlayerService: playerState collect update | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:56.973  1248  1248 W VideoPlayerService: updateLocks active=false | exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 wakeHeld=true wifiHeld=true ongoing=true activeForLocks=false
08-18 11:42:56.973  1248  1248 E PlayerErrorHandler: Player error (notify): Source error
08-18 11:42:56.974  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=1 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:56.977  1248  1248 W FlowVideoAutoNext: onTimelineChanged reason=0 windows=0 | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:56.977  1248  1248 W FlowVideoAutoNext: onMediaItemTransition reason=3 mediaId=null | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:59.120  1248  1248 W EnhancedPlayerManager: Critical memory pressure; releasing video-heavy player state
08-18 11:42:59.122  1248  1248 W FlowVideoAutoNext: switchToAudioOnly | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=false live=false
08-18 11:42:59.122  1248  1248 W FlowVideoAutoNext: setVideoTracksDisabled disabled=true | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:59.122  1248  1248 W FlowVideoAutoNext: schedulePreloadNext skipped no current media item | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:59.826  1248  1248 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
08-18 11:42:59.826  1248  1248 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
08-18 11:42:59.826  1248  1248 W FlowVideoLifecycle: onResume | interactive=true lifecycle=STARTED pip=false pendingAutoPip=false bgPref=true shortsBgPref=false explicitBg=false video=15cEtTMmvJk exo=IDLE pwr=true playing=false buffering=false pos=0/-9223372036854775807 idx=0 count=0
08-18 11:42:59.832  1248  1248 W FlowVideoAutoNext: restoreVideoOutput deferred interactive=true surfaceValid=false | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=true tracksDisabled=true surface=false live=false
08-18 11:42:59.835  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:null
08-18 11:42:59.837  1248  1248 W FlowVideoLifecycle: surfaceAttached video=15cEtTMmvJk pos=0 pwr=true force=true
08-18 11:42:59.857  1248  1248 W FlowVideoAutoNext: restoreVideoOutput | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=true surface=true live=false
08-18 11:42:59.857  1248  1248 W FlowVideoAutoNext: setVideoTracksDisabled disabled=false | video=15cEtTMmvJk exo=IDLE pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=true live=false
08-18 11:42:59.861  1248  1248 W VRI[IconFlowRed]: handleResized, msg:, frameChanged:false, configChanged:false, displayChanged:false, attachedFrameChanged:false, compatScaleChanged:false, pendingDragResizing=false
08-18 11:42:59.862  1248  1248 W VRI[IconFlowRed]: updateBlastSurfaceIfNeeded, surfaceSize:Point(1272, 2772), lastSurfaceSize:Point(1272, 2772), format:-3, blastBufferQueue:android.graphics.BLASTBufferQueue@e8d47b7
08-18 11:43:02.524  1248  1248 W PlayerErrorHandler: Refocus: player IDLE — calling prepare()
08-18 11:43:02.524  1248  1248 W FlowVideoAutoNext: onPlaybackStateChanged ENDED recovering=true live=false | video=15cEtTMmvJk exo=ENDED pwr=true playing=false pos=0/-9223372036854775807 idx=0 count=0 queue=0/113 hasNext=true autoplay=false candidates=20 target=JgeKzO5xSUE preloaded=null attempt=null->null retry=0 audioOnly=false tracksDisabled=false surface=true live=false

--- a/app/src/main/assets/changelog/v2.2.1.txt
+++ b/app/src/main/assets/changelog/v2.2.1.txt
@@ -22,6 +22,7 @@ IMPROVEMENTS
 - Scale the long-press speed boost from the current speed so it can never slow playback down #867
 - Restyle the long-press speed indicator as a pill showing the speed it will actually apply
 - Ripple the tapped side of the screen when double-tap seeking
+- Offer every dubbed audio track and the full quality ladder on videos that provide them
 
 PERFORMANCE
 - Faster cold start and smoother scrolling
@@ -61,6 +62,7 @@ FIXES AND STABILITY
 - Fix a crash when playback notification artwork finished loading in the background
 - Fix the playback notification disappearing when the queue auto-advanced in the background
 - Fix videos cutting out around the one-minute mark on auto-dubbed titles
+- Fix playback stopping about a minute in and then retrying until it gave up #921 #922
 - Fix the wrong audio track being requested on auto-dubbed videos
 - Fix the mini player expanding back to the full player while metadata was still loading
 - Fix playback on videos that only offer a server-side adaptive stream
@@ -74,7 +76,7 @@ FIXES AND STABILITY
 - Fix swiping down to leave fullscreen committing even after you dragged back up
 
 DEPENDENCY
-- Bump NewPipeExtractor to v0.26.4
+- Bump NewPipeExtractor to v0.26.5
 - Update Gradle to 9.6.1 and Android Gradle Plugin to 9.3.1
 - Update Kotlin to 2.4.10 with the Compose compiler plugin and KSP 2.3.11
 - Update Hilt to 2.60.1 and Room to 2.8.4

--- a/app/src/main/java/io/github/aedev/flow/innertube/models/YouTubeClient.kt
+++ b/app/src/main/java/io/github/aedev/flow/innertube/models/YouTubeClient.kt
@@ -2,6 +2,26 @@ package io.github.aedev.flow.innertube.models
 
 import kotlinx.serialization.Serializable
 
+/**
+ * Which attestation runtime mints a PO Token this client's requests will be accepted with.
+ *
+ * A PO Token is produced by BotGuard (Web), DroidGuard (Android) or iOSGuard (iOS), and one
+ * platform's token is never valid on another's URL. Flow only owns a BotGuard WebView, so [WEB]
+ * is the only family it can attest; the rest are usable exactly as long as YouTube keeps serving
+ * them unattested. Encoding that here is what stops a WEB token being stamped onto an
+ * ANDROID_VR/IOS URL, which is a claim GVS checks and rejects.
+ */
+enum class AttestationPlatform {
+    /** BotGuard — Flow mints these in [io.github.aedev.flow.utils.potoken.PoTokenWebView]. */
+    WEB,
+
+    /** DroidGuard, inside Google Play Services. Not reachable from a third-party app. */
+    DROIDGUARD,
+
+    /** iOSGuard. Not reachable at all from Android. */
+    IOSGUARD,
+}
+
 @Serializable
 data class YouTubeClient(
     val clientName: String,
@@ -25,6 +45,12 @@ data class YouTubeClient(
     val useSignatureTimestamp: Boolean = false,
     val isEmbedded: Boolean = false,
     val useWebPoTokens: Boolean = false,
+    /**
+     * The runtime that mints tokens this client's URLs will accept. Defaults to [AttestationPlatform.WEB]
+     * because every client Flow can actually attest is web-family; the non-web clients set it
+     * explicitly so token attachment can be gated on it.
+     */
+    val attestation: AttestationPlatform = AttestationPlatform.WEB,
     /**
      * Whether [userAgent] is repeated inside `context.client` as well as sent as the HTTP header.
      * Only MWEB expects this; sending it for other clients is a fingerprint inconsistency. Clients
@@ -121,6 +147,7 @@ data class YouTubeClient(
                 friendlyName = "Android",
                 loginSupported = true,
                 useSignatureTimestamp = true,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
         val WEB_REMIX =
@@ -180,6 +207,7 @@ data class YouTubeClient(
                 clientId = "5",
                 userAgent = "com.google.ios.youtube/21.03.1 (iPhone16,2; U; CPU iOS 18_2 like Mac OS X;)",
                 osVersion = "18.2.22C152",
+                attestation = AttestationPlatform.IOSGUARD,
             )
 
         val MOBILE =
@@ -190,6 +218,7 @@ data class YouTubeClient(
                 userAgent = "com.google.android.youtube/21.03.38 (Linux; U; Android 14) gzip",
                 loginSupported = true,
                 useSignatureTimestamp = true,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
         val ANDROID_VR_NO_AUTH =
@@ -202,6 +231,7 @@ data class YouTubeClient(
                         "(Linux; U; Android 12; en_US; Oculus Quest 3; Build/SQ3A.220605.009.A1; Cronet/132.0.6808.3)",
                 loginSupported = false,
                 useSignatureTimestamp = false,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
         val ANDROID_VR_1_61_48 =
@@ -223,6 +253,7 @@ data class YouTubeClient(
                 friendlyName = "Android VR 1.61",
                 loginSupported = false,
                 useSignatureTimestamp = false,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
         val ANDROID_VR_1_65_10 =
@@ -243,6 +274,7 @@ data class YouTubeClient(
                 friendlyName = "Android VR 1.65",
                 loginSupported = false,
                 useSignatureTimestamp = false,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
         val ANDROID_VR_1_43_32 =
@@ -264,6 +296,7 @@ data class YouTubeClient(
                 friendlyName = "Android VR 1.43",
                 loginSupported = false,
                 useSignatureTimestamp = false,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
         val ANDROID_CREATOR =
@@ -285,23 +318,36 @@ data class YouTubeClient(
                 friendlyName = "Android Studio",
                 loginSupported = true,
                 useSignatureTimestamp = true,
+                attestation = AttestationPlatform.DROIDGUARD,
             )
 
+        /**
+         * The primary direct-URL client. GVS serves its formats without a PO Token and without an
+         * `n` parameter, so it reaches first frame with no attestation and no nsig decode — the two
+         * properties ANDROID_VR was chosen for, which ANDROID_VR lost when YouTube began requiring
+         * a GVS PO Token for it (yt-dlp #17261, Aug 2026).
+         *
+         * Version-sensitive: the previous `0.1`/`RealityDevice14,1` build still answers, but serves
+         * 17 formats and a single audio track. This build serves 98 formats and every dubbed track,
+         * so do not "simplify" these values — they are the difference between a full ladder and a
+         * stub. Kept in step with yt-dlp's `visionos` entry in `youtube/_base.py`.
+         */
         val VISIONOS =
             YouTubeClient(
                 clientName = "VISIONOS",
-                clientVersion = "0.1",
+                clientVersion = "1.02",
                 clientId = "101",
                 userAgent =
-                    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 " +
-                        "(KHTML, like Gecko) Version/18.0 Safari/605.1.15",
+                    "Mozilla/5.0 (Macintosh; Intel Mac OS X 15_7_3) AppleWebKit/605.1.15 " +
+                        "(KHTML, like Gecko) Version/26.0 Safari/605.1.15",
                 osName = "visionOS",
-                osVersion = "1.3.21O771",
+                osVersion = "26.5.23O471",
                 deviceMake = "Apple",
-                deviceModel = "RealityDevice14,1",
+                deviceModel = "RealityDevice17,1",
                 friendlyName = "visionOS",
                 loginSupported = false,
                 useSignatureTimestamp = false,
+                attestation = AttestationPlatform.IOSGUARD,
             )
 
         val IPADOS =
@@ -318,6 +364,7 @@ data class YouTubeClient(
                 loginSupported = false,
                 useSignatureTimestamp = false,
                 packageName = "com.google.ios.youtube",
+                attestation = AttestationPlatform.IOSGUARD,
             )
     }
 }

--- a/app/src/main/java/io/github/aedev/flow/player/stream/InnerTubeVideoStreamExtractor.kt
+++ b/app/src/main/java/io/github/aedev/flow/player/stream/InnerTubeVideoStreamExtractor.kt
@@ -4,6 +4,7 @@ import android.net.Uri
 import android.util.Log
 import androidx.media3.common.util.UnstableApi
 import io.github.aedev.flow.innertube.YouTube
+import io.github.aedev.flow.innertube.models.AttestationPlatform
 import io.github.aedev.flow.innertube.models.YouTubeClient
 import io.github.aedev.flow.innertube.models.YouTubeLocale
 import io.github.aedev.flow.innertube.models.response.PlayerResponse
@@ -51,15 +52,12 @@ object InnerTubeVideoStreamExtractor {
         val forceSabr: Boolean,
     )
 
-    // * Fast, token-free clients tried first. They return direct adaptive URLs (played via normal DASH/progressive) when not bot-walled
+    // The token-free direct client. VISIONOS alone: it is the only client that still serves direct
+    // adaptive URLs GVS will honour for the whole video without a PO Token, and it does so without
+    // an `n` parameter, so first frame costs neither an attestation nor an nsig decode.
     private val FAST_CLIENTS: List<YouTubeClient> =
         listOf(
-            YouTubeClient.ANDROID_VR_1_61_48,
-            YouTubeClient.ANDROID_VR_NO_AUTH,
-            YouTubeClient.ANDROID_VR_1_65_10,
-            YouTubeClient.IPADOS,
-            YouTubeClient.IOS,
-            YouTubeClient.ANDROID_VR_1_43_32,
+            YouTubeClient.VISIONOS,
         )
 
     private val BOT_RESISTANT_CLIENTS: List<YouTubeClient> =
@@ -67,23 +65,38 @@ object InnerTubeVideoStreamExtractor {
             YouTubeClient.TVHTML5_SIMPLY_EMBEDDED_PLAYER,
         )
 
-    // Last-resort token-free clients tried after the durable WEB+SABR path
+    /**
+     * Direct clients GVS now cuts off after ~60s of media without a PO Token Flow cannot mint
+     * (DroidGuard). Kept below the attested SABR path rather than deleted: they still answer, still
+     * carry a full ladder, and are the difference between degraded playback and none when both
+     * VISIONOS and SABR are unavailable. Never promote these above [SABR_CLIENTS].
+     */
+    private val GATED_FALLBACK_CLIENTS: List<YouTubeClient> =
+        listOf(
+            YouTubeClient.ANDROID_VR_1_61_48,
+            YouTubeClient.ANDROID_VR_NO_AUTH,
+            YouTubeClient.ANDROID_VR_1_65_10,
+            YouTubeClient.ANDROID_VR_1_43_32,
+        )
+
+    // Last-resort token-free clients, tried after everything else
     private val LAST_RESORT_CLIENTS: List<YouTubeClient> =
         listOf(
             YouTubeClient.MOBILE,
             YouTubeClient.ANDROID_CREATOR,
         )
 
+    // MWEB first: it carries every dubbed audio track and is web-family, so its GVS token is one
+    // Flow's own BotGuard session can mint.
     private val SABR_CLIENTS: List<YouTubeClient> =
         listOf(
-            YouTubeClient.WEB,
             YouTubeClient.MWEB,
+            YouTubeClient.WEB,
         )
 
     private val LIVE_MANIFEST_CLIENTS: List<YouTubeClient> =
         listOf(
-            YouTubeClient.IOS,
-            YouTubeClient.IPADOS,
+            YouTubeClient.VISIONOS,
             YouTubeClient.TVHTML5_SIMPLY_EMBEDDED_PLAYER,
             YouTubeClient.ANDROID_VR_1_61_48,
         )
@@ -139,12 +152,6 @@ object InnerTubeVideoStreamExtractor {
                 return@withContext result
             }
 
-            tryDirectClients(videoId, BOT_RESISTANT_CLIENTS, failureReasons, liveDetected = liveDetected)?.let { direct ->
-                val result = maybeUpgradeToSabr(videoId, direct, failureReasons)
-                Log.w(TAG, "Extraction OK for $videoId via ${result.usedClient.clientName} (mode=${resultMode(result)})")
-                return@withContext result
-            }
-
             if (liveDetected[0]) {
                 tryLiveClients(videoId, failureReasons)?.let {
                     Log.w(TAG, "Live manifest for $videoId via ${it.usedClient.clientName} (live-clients)")
@@ -152,14 +159,35 @@ object InnerTubeVideoStreamExtractor {
                 }
             }
 
-            // 2) Durable path: web client + BotGuard PoToken + SABR. Survives the LOGIN_REQUIRED bot wall
+            // 2) Durable path: web client + BotGuard PoToken + SABR. Ranked above every remaining
+            // direct client because it is the only other path whose token Flow can actually mint —
+            // the clients below it are all served unattested and can be cut off mid-playback.
             trySabrClients(videoId, failureReasons)?.let {
                 Log.w(TAG, "Extraction OK for $videoId via ${it.usedClient.clientName} (mode=SABR)")
                 PlayerDiagnostics.logWarning(TAG, "extract OK $videoId mode=SABR (durable) via ${it.usedClient.clientName}")
                 return@withContext if (liveDetected[0] && !it.isLive) it.copy(isLive = true) else it
             }
 
-            // 3) Last resort: remaining token-free clients
+            tryDirectClients(videoId, BOT_RESISTANT_CLIENTS, failureReasons, liveDetected = liveDetected)?.let { direct ->
+                val result = maybeUpgradeToSabr(videoId, direct, failureReasons)
+                Log.w(TAG, "Extraction OK for $videoId via ${result.usedClient.clientName} (mode=${resultMode(result)})")
+                return@withContext result
+            }
+
+            // 3) Gated direct clients. Playable, but GVS stops serving them ~60s in, so they rank
+            // below anything attested and are only reached when the paths above are unavailable.
+            tryDirectClients(videoId, GATED_FALLBACK_CLIENTS, failureReasons, liveDetected = liveDetected)?.let { direct ->
+                val result = maybeUpgradeToSabr(videoId, direct, failureReasons)
+                Log.w(TAG, "Extraction OK for $videoId via ${result.usedClient.clientName} (mode=${resultMode(result)}/gated)")
+                PlayerDiagnostics.logWarning(
+                    TAG,
+                    "extract OK $videoId via ${result.usedClient.clientName} mode=${resultMode(result)}/GATED — " +
+                        "unattested, GVS may stop serving ~60s in",
+                )
+                return@withContext result
+            }
+
+            // 4) Last resort: remaining token-free clients
             tryDirectClients(videoId, LAST_RESORT_CLIENTS, failureReasons, allowUntransformedN = true, liveDetected = liveDetected)?.let {
                 Log.w(TAG, "Extraction OK for $videoId via ${it.usedClient.clientName} (mode=DIRECT/last-resort)")
                 PlayerDiagnostics.logWarning(TAG, "extract OK $videoId via ${it.usedClient.clientName} mode=DIRECT/last-resort")
@@ -287,14 +315,13 @@ object InnerTubeVideoStreamExtractor {
                 try {
                     Log.d(TAG, "Trying ${client.clientName} v${client.clientVersion}")
 
-                    // ANDROID_VR's /player request must be called WITHOUT a PoToken: injecting the
-                    // WEB-bound BotGuard token makes YT reject the request (non-OK / empty
-                    // streamingData), which was silently dropping playback onto the slower clients.
-                    // Its URLs still need GVS protection — that is the `pot=` URL param attached to
-                    // the winner's formats below, not the /player token. Other fast clients keep the
-                    // /player attestation.
-                    val isAndroidVr = client.clientName == "ANDROID_VR"
-                    val clientPoToken = if (isAndroidVr) null else awaitMint()?.playerRequestPoToken
+                    // Only web-family clients get the BotGuard token. Injecting it into an
+                    // ANDROID_VR/IOS /player request makes YT reject it outright (non-OK / empty
+                    // streamingData), and stamping it on their URLs is worse than sending nothing:
+                    // it is a claim GVS validates and refuses. Those clients are usable only for as
+                    // long as YouTube serves them unattested.
+                    val webAttested = client.attestation == AttestationPlatform.WEB
+                    val clientPoToken = if (webAttested) awaitMint()?.playerRequestPoToken else null
 
                     val playerResponse =
                         withTimeoutOrNull(PER_CLIENT_TIMEOUT_MS) {
@@ -405,8 +432,12 @@ object InnerTubeVideoStreamExtractor {
                     )
 
                     val urlPot =
-                        withTimeoutOrNull(STREAM_POT_ATTACH_GRACE_MS) { awaitMint() }?.streamingDataPoToken
-                    if (urlPot == null) {
+                        if (webAttested) {
+                            withTimeoutOrNull(STREAM_POT_ATTACH_GRACE_MS) { awaitMint() }?.streamingDataPoToken
+                        } else {
+                            null
+                        }
+                    if (urlPot == null && webAttested) {
                         PlayerDiagnostics.logWarning(
                             TAG,
                             "${client.clientName} winner for $videoId shipping pot-less (mint not ready in " +
@@ -524,8 +555,15 @@ object InnerTubeVideoStreamExtractor {
                 return null
             }
 
-            // StreamerContext.po_token carries the videoId-bound content token. The visitor-bound streaming token here draws
-            // sabr.media_serving_enforcement_id_error from GVS.
+            // StreamerContext.po_token carries the videoId-bound content token, which is what GVS
+            // accepts here in practice.
+            //
+            // The visitor-bound token used to draw sabr.media_serving_enforcement_id_error, and the
+            // reason is now understood: it was minted against the whole visitorData blob rather than
+            // the 11-char visitor ID GVS actually binds to (see [VisitorId]). That binding is fixed,
+            // so the streaming token is a candidate here too — but this path is the durable fallback
+            // and the videoId-bound token is the one measured to work, so switching wants an
+            // on-device A/B rather than an assumption.
             val resolved =
                 if (targetHeight > 0) {
                     SabrUrlResolver.resolveForQuality(
@@ -652,13 +690,14 @@ object InnerTubeVideoStreamExtractor {
         return null
     }
 
-    private fun PlayerResponse.isLiveNow(): Boolean {
-        val details = videoDetails
-        return details?.isLive == true ||
-            details?.isPostLiveDvr == true ||
-            playabilityStatus.liveStreamability != null ||
-            !streamingData?.hlsManifestUrl.isNullOrBlank()
-    }
+    private fun PlayerResponse.isLiveNow(): Boolean =
+        LiveDetectionRules.isLiveNow(
+            isLive = videoDetails?.isLive,
+            isPostLiveDvr = videoDetails?.isPostLiveDvr,
+            hasLiveStreamability = playabilityStatus.liveStreamability != null,
+            hasHlsManifest = !streamingData?.hlsManifestUrl.isNullOrBlank(),
+            hasAdaptiveFormats = !streamingData?.adaptiveFormats.isNullOrEmpty(),
+        )
 
     private fun PlayerResponse.toLiveResultOrNull(client: YouTubeClient): VideoExtractionResult? {
         val hls = streamingData?.hlsManifestUrl?.takeIf { it.isNotBlank() }

--- a/app/src/main/java/io/github/aedev/flow/player/stream/LiveDetectionRules.kt
+++ b/app/src/main/java/io/github/aedev/flow/player/stream/LiveDetectionRules.kt
@@ -1,0 +1,28 @@
+package io.github.aedev.flow.player.stream
+
+/**
+ * Decides whether a player response should be played as a live stream rather than as a VOD ladder.
+ *
+ * Kept pure and separate because getting it wrong is silent and expensive: a VOD misread as live is
+ * handed to the player with no video/audio formats at all, so quality and audio-track selection come
+ * up empty rather than failing loudly.
+ */
+object LiveDetectionRules {
+    /**
+     * @param hasHlsManifest whether `streamingData.hlsManifestUrl` is present. Deliberately weak
+     *   evidence: Apple-family clients (VISIONOS, IOS) return a manifest for ordinary VODs as well
+     *   as for live, so it only implies live when there is no adaptive ladder to play instead.
+     * @param hasAdaptiveFormats whether the response carries any adaptive formats.
+     */
+    fun isLiveNow(
+        isLive: Boolean?,
+        isPostLiveDvr: Boolean?,
+        hasLiveStreamability: Boolean,
+        hasHlsManifest: Boolean,
+        hasAdaptiveFormats: Boolean,
+    ): Boolean {
+        if (isLive == true || isPostLiveDvr == true) return true
+        if (hasLiveStreamability) return true
+        return hasHlsManifest && !hasAdaptiveFormats
+    }
+}

--- a/app/src/main/java/io/github/aedev/flow/utils/MusicPlayerUtils.kt
+++ b/app/src/main/java/io/github/aedev/flow/utils/MusicPlayerUtils.kt
@@ -11,11 +11,10 @@ import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.ANDROID_CRE
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.ANDROID_VR_1_43_32
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.ANDROID_VR_1_61_48
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.ANDROID_VR_NO_AUTH
-import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.IOS
-import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.IPADOS
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.MOBILE
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.TVHTML5
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.TVHTML5_SIMPLY_EMBEDDED_PLAYER
+import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.VISIONOS
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.WEB
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.WEB_CREATOR
 import io.github.aedev.flow.innertube.models.YouTubeClient.Companion.WEB_REMIX
@@ -62,22 +61,25 @@ object MusicPlayerUtils {
             ANDROID_VR_1_43_32,
             ANDROID_VR_1_61_48,
             ANDROID_CREATOR,
-            IPADOS,
             ANDROID_VR_NO_AUTH,
             MOBILE,
-            IOS,
             WEB,
             WEB_CREATOR,
         )
 
+    /**
+     * Tried before [STREAM_FALLBACK_CLIENTS]. VISIONOS leads because it is the only direct client
+     * GVS still serves past ~60s without a PO Token — the rest are unattested and get cut off
+     * mid-track, which is what stopped playlists at 59 seconds. IOS/IPADOS are gone entirely: they
+     * now return SABR-only responses with no direct URLs at all.
+     */
     private val FAST_DIRECT_STREAM_CLIENTS: Array<YouTubeClient> =
         arrayOf(
+            VISIONOS,
             ANDROID_VR_1_43_32,
             ANDROID_VR_1_61_48,
             ANDROID_VR_NO_AUTH,
-            IPADOS,
             MOBILE,
-            IOS,
             ANDROID_CREATOR,
         )
 
@@ -488,10 +490,13 @@ object MusicPlayerUtils {
             return null
         }
 
+        // Direct-URL clients skip the probe: their URLs are playable as returned, and the extra
+        // round trip would sit on the critical path to first audio.
         val needsValidation =
             validate &&
                 !client.clientName.startsWith("ANDROID") &&
-                client.clientName != "IOS"
+                client.clientName != "IOS" &&
+                client.clientName != "VISIONOS"
         if (needsValidation && !checkUrl(resolved.url, client.userAgent)) {
             Log.d(TAG, "URL validation failed for ${client.clientName}")
             return null

--- a/app/src/main/java/io/github/aedev/flow/utils/potoken/PoTokenGenerator.kt
+++ b/app/src/main/java/io/github/aedev/flow/utils/potoken/PoTokenGenerator.kt
@@ -148,10 +148,14 @@ object PoTokenGenerator {
                 // GVS honors cold/low-trust attestations only briefly (the mid-playback 403).
                 // Re-run the full BotGuard challenge until the minted token reaches the
                 // documented 110-128 byte range, like the desktop minter's retry loop.
+                // Bound to the 11-char visitor ID, not the visitorData blob it sits inside: GVS
+                // validates the streaming token against the ID, so minting against the blob yields
+                // a token that is accepted at load time and refused once enforcement kicks in.
+                val gvsBinding = VisitorId.gvsBinding(sessionId)
                 var attempt = 0
                 while (attempt < STREAMING_POT_ATTEMPTS) {
                     val generator = attestedGenerator()
-                    val pot = generator.generatePoToken(sessionId)
+                    val pot = generator.generatePoToken(gvsBinding)
                     newStreamingPot = pot
                     lowTrust = PoTokenAttestationPolicy.isLowTrust(pot)
                     if (!lowTrust) break

--- a/app/src/main/java/io/github/aedev/flow/utils/potoken/VisitorId.kt
+++ b/app/src/main/java/io/github/aedev/flow/utils/potoken/VisitorId.kt
@@ -1,0 +1,44 @@
+package io.github.aedev.flow.utils.potoken
+
+import java.net.URLDecoder
+import java.util.Base64
+
+/**
+ * Extracts the 11-character visitor ID that a GVS PoToken must be bound to.
+ *
+ * `visitorData` is a base64url protobuf several hundred characters long, and its first field is the
+ * short visitor ID. GVS binds a logged-out streaming token to **that ID**, not to the surrounding
+ * blob — binding to the blob produces a token the server rejects (`sabr.media_serving_enforcement_id_error`
+ * on the SABR path, a plain 403 past the free window on direct URLs).
+ *
+ * Mirrors yt-dlp's `_extract_visitor_id` in `youtube/pot/utils.py`, including its default of
+ * preferring the visitor ID over the raw blob.
+ */
+object VisitorId {
+    /** Field 1 (`0x0A`), length 11 (`0x0B`) — the two header bytes before the ID itself. */
+    private const val ID_START = 2
+    private const val ID_LENGTH = 11
+    private val ID_PATTERN = Regex("[A-Za-z0-9_-]{$ID_LENGTH}")
+
+    /**
+     * The visitor ID inside [visitorData], or null when it cannot be read — a caller that gets null
+     * should bind to [visitorData] unchanged rather than skip attestation, matching yt-dlp's fallback.
+     */
+    fun extract(visitorData: String?): String? {
+        if (visitorData.isNullOrBlank()) return null
+        return try {
+            // Padding is optional in the wild, so decode without it and add our own.
+            val decoded = URLDecoder.decode(visitorData, Charsets.UTF_8.name()).trimEnd('=')
+            val padded = decoded.padEnd(decoded.length + (4 - decoded.length % 4) % 4, '=')
+            val bytes = Base64.getUrlDecoder().decode(padded)
+            if (bytes.size < ID_START + ID_LENGTH) return null
+            val id = String(bytes, ID_START, ID_LENGTH, Charsets.UTF_8)
+            id.takeIf { ID_PATTERN.matches(it) }
+        } catch (e: IllegalArgumentException) {
+            null
+        }
+    }
+
+    /** The identifier a GVS/streaming token should be minted against for this session. */
+    fun gvsBinding(visitorData: String): String = extract(visitorData) ?: visitorData
+}

--- a/app/src/test/java/io/github/aedev/flow/player/stream/LiveDetectionRulesTest.kt
+++ b/app/src/test/java/io/github/aedev/flow/player/stream/LiveDetectionRulesTest.kt
@@ -1,0 +1,64 @@
+package io.github.aedev.flow.player.stream
+
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+/**
+ * Cases are the real shapes observed from the InnerTube `player` endpoint, not invented ones —
+ * every VISIONOS response carries an HLS manifest, which is what made the naive rule misfire.
+ */
+class LiveDetectionRulesTest {
+    private fun rule(
+        isLive: Boolean? = null,
+        isPostLiveDvr: Boolean? = null,
+        hasLiveStreamability: Boolean = false,
+        hasHlsManifest: Boolean = false,
+        hasAdaptiveFormats: Boolean = false,
+    ) = LiveDetectionRules.isLiveNow(
+        isLive = isLive,
+        isPostLiveDvr = isPostLiveDvr,
+        hasLiveStreamability = hasLiveStreamability,
+        hasHlsManifest = hasHlsManifest,
+        hasAdaptiveFormats = hasAdaptiveFormats,
+    )
+
+    @Test
+    fun `visionos VOD ships an hls manifest and is not live`() {
+        // The regression: 98 adaptive formats alongside a manifest. Reading this as live returns an
+        // empty ladder, so codecs and audio tracks vanish from the UI.
+        assertFalse(rule(hasHlsManifest = true, hasAdaptiveFormats = true))
+    }
+
+    @Test
+    fun `visionos short ships an hls manifest and is not live`() {
+        assertFalse(rule(hasHlsManifest = true, hasAdaptiveFormats = true))
+    }
+
+    @Test
+    fun `a broadcasting stream is live`() {
+        assertTrue(rule(isLive = true, hasLiveStreamability = true, hasHlsManifest = true, hasAdaptiveFormats = true))
+    }
+
+    @Test
+    fun `a post-live dvr stream is live even with an adaptive ladder`() {
+        assertTrue(rule(isPostLiveDvr = true, hasHlsManifest = true, hasAdaptiveFormats = true))
+    }
+
+    @Test
+    fun `live streamability alone is enough`() {
+        assertTrue(rule(hasLiveStreamability = true))
+    }
+
+    @Test
+    fun `a manifest with nothing else playable is still treated as live`() {
+        // The clause's original purpose, preserved: SABR-only live responses expose no adaptive
+        // formats, so the manifest is the only thing there is to play.
+        assertTrue(rule(hasHlsManifest = true, hasAdaptiveFormats = false))
+    }
+
+    @Test
+    fun `a plain VOD with neither manifest nor live flags is not live`() {
+        assertFalse(rule(hasAdaptiveFormats = true))
+    }
+}

--- a/app/src/test/java/io/github/aedev/flow/utils/potoken/VisitorIdTest.kt
+++ b/app/src/test/java/io/github/aedev/flow/utils/potoken/VisitorIdTest.kt
@@ -1,0 +1,44 @@
+package io.github.aedev.flow.utils.potoken
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Test
+
+class VisitorIdTest {
+    /** A real visitorData captured from youtube.com; its visitor ID is `rlqDISqy6-0`. */
+    private val realVisitorData =
+        "CgtybHFESVNxeTYtMCjA9pLUBjIKCgJMQhIEGgAgP2LfAgrcAjIxLllUPWhpRGRwQmZwWFRUNU9WRkJT"
+
+    @Test
+    fun `extracts the visitor id from real visitor data`() {
+        assertEquals("rlqDISqy6-0", VisitorId.extract(realVisitorData))
+    }
+
+    @Test
+    fun `url-encoded padding does not defeat extraction`() {
+        assertEquals("rlqDISqy6-0", VisitorId.extract("$realVisitorData%3D%3D"))
+    }
+
+    @Test
+    fun `gvsBinding prefers the visitor id over the blob`() {
+        val binding = VisitorId.gvsBinding(realVisitorData)
+        assertEquals("rlqDISqy6-0", binding)
+        // The whole point: GVS binds to the short id, not the several-hundred-character blob.
+        assertEquals(11, binding.length)
+    }
+
+    @Test
+    fun `gvsBinding falls back to the raw value when no id can be read`() {
+        val notProtobuf = "not-base64-at-all!!"
+        assertEquals(notProtobuf, VisitorId.gvsBinding(notProtobuf))
+    }
+
+    @Test
+    fun `blank and malformed input yield null`() {
+        assertNull(VisitorId.extract(null))
+        assertNull(VisitorId.extract(""))
+        assertNull(VisitorId.extract("   "))
+        // Valid base64, but too short to contain an id.
+        assertNull(VisitorId.extract("CgQ="))
+    }
+}

